### PR TITLE
Add UserResource to OAuth Monolith

### PR DESCRIPTION
**Pull request by mraible**
Adding `UserResource` allows many-to-one relations to the `User` object.

- [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/server/templates/src/test/java/package/web/rest/_UserResourceIntTest.java [hunk 1]
```diff
@@ -63,10 +64,7 @@
 <%_ if (databaseType !== 'cassandra') { _%>
 import java.time.Instant;
 <%_ } _%>
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
```
Comment (mraible) [inline generators/server/templates/src/test/java/package/web/rest/_UserResourceIntTest.java]: `java.util.Arrays` and `java.util.List` is not used with OAuth; it seemed easier to import `java.util.*` than have if statements for each import.

**Comment by cbornet** [comment]
Since this PR, the test context is restarted by AccountResourceIntTest for app-ng2-oauth :disappointed: : https://travis-ci.org/jhipster/generator-jhipster/jobs/289920669#L2381

**Comment by jdubois** [comment]
I also have the `AccountResourceIntTest` which does not even compile on a OAuth2 microservice - there's definitely something broken here

**Comment by jdubois** [comment]
Same thing for an OAuth2 gateway... I'm doing a specific ticket for all this, as this is getting really big. We don't test those in our build, for as far as I'm concerned this means the build is broken and we can't do a release now (and I wanted to do one for @xetys )

**Pull request merged**